pkgs/by-name/fi/firefly-iii-data-importer/package.nix

```diff
@@ -0,0 +1,92 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenvNoCC,
+  nodejs,
+  fetchNpmDeps,
+  buildPackages,
+  php83,
+  nixosTests,
+  nix-update-script,
+  dataDir ? "/var/lib/firefly-iii-data-importer",
+}:
+
+let
+  pname = "firefly-iii-data-importer";
+  version = "1.5.3";
+  phpPackage = php83;
+  npmDepsHash = "sha256-AdSPHK9Ws/VMqzuvfbShdAl1va65G1K0N379JS4Az5E=";
+
+  src = fetchFromGitHub {
+    owner = "firefly-iii";
+    repo = "data-importer";
+    rev = "v${version}";
+    hash = "sha256-5hIH/XDyPn6CokqEO/35Xu+K9KTVYyC2jxADCU69uxI=";
+  };
+in
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  inherit pname src version;
+
+  buildInputs = [ phpPackage ];
+
+  nativeBuildInputs = [
+    nodejs
+    nodejs.python
+    buildPackages.npmHooks.npmConfigHook
+    phpPackage.composerHooks.composerInstallHook
+    phpPackage.packages.composer-local-repo-plugin
+  ];
+
+  composerNoDev = true;
+  composerNoPlugins = true;
+  composerNoScripts = true;
+  composerStrictValidation = true;
+  strictDeps = true;
+
+  vendorHash = "sha256-uClTcT6qZSN2UNop4+T0c3GNozASCUk6enMfYPmzsz8=";
+
+  npmDeps = fetchNpmDeps {
+    inherit src;
+    name = "${pname}-npm-deps";
+    hash = npmDepsHash;
+  };
+
+  composerRepository = phpPackage.mkComposerRepository {
+    inherit (finalAttrs)
+      pname
+      src
+      vendorHash
+      version
+      ;
+    composerNoDev = true;
+    composerNoPlugins = true;
+    composerNoScripts = true;
+    composerStrictValidation = true;
+  };
+
+  preInstall = ''
+    npm run build --workspace=v2
+  '';
+
+  passthru = {
+    inherit phpPackage;
+    tests = nixosTests.firefly-iii-data-importer;
+    updateScript = nix-update-script { };
+  };
+
+  postInstall = ''
+    mv $out/share/php/${pname}/* $out/
+    rm -R $out/share $out/storage $out/bootstrap/cache $out/node_modules
```

Comment:
It would be nicer to not copy all those files in the first place since that is likely across mounts.

Reply:
Apart from `node_modules`, the others are empty directories, mostly. `node_modules` is generated by the install hook. I'm not sure what you mean by not copying, sorry.

Reply:
@savyajha I think what @SuperSandro2000  meant is to first delete the files and then move them

Reply:
@bachp Made the changes